### ISSUE TITLE
Add minimum required styles for CSS transitions

### ISSUE THREAD

**Issue by smillart** [smillart/WAI-ARIA-Patterns-And-Widgets#2]
Provide default styles for accordion panels where CSS transitions can be applied from hidden to revealed state (and vice versa).

**Comment by laurent-b4y** [comment]
`.aria-accordion .aria-accordion__panel--transition[hidden] {
  display: block;
  height: auto;
  max-height: 0;
  overflow: hidden;
  -webkit-transition: max-height 800ms ease-out;
  -moz-transition: max-height 800ms ease-out;
  -ms-transition: max-height 800ms ease-out;
  -o-transition: max-height 800ms ease-out;
  transition: max-height 800ms ease-out;
  visibility: visible;
  /* persist visibility value from animation */
  animation: 800ms delay-visibility;
}
.aria-accordion .aria-accordion__panel--transition {
  display: block;
  max-height: 1000px;
  visibility: visible;
  -webkit-transition: max-height 800ms ease-in;
  -moz-transition: max-height 800ms ease-in;
  -ms-transition: max-height 800ms ease-in;
  -o-transition: max-height 800ms ease-in;
  transition: max-height 800ms ease-in;
  overflow: visible;
  /* persist overflow value from animation */
  animation: 800ms delay-overflow;
}
@keyframes delay-visibility {
  from { visibility: visible; }
}
@keyframes delay-overflow {
  from { overflow: hidden; }
}`

This is what I used for the ckeditor module version in order to have some animation.

**Comment by smillart** [comment]
Thank you Laurent for your input!

Given the use of BEM (**B**lock/**E**lement/**M**odifier) methodology `.block-name__element-name--modifier-name` for CSS class naming convention, nested selectors produces redundant code. Using single selector is suffisant:

**BEM Sass usage:**
```scss
// Block name: Accordion
//
.aria-accordion {

  // Element name: Accordion panel
  //
  &__panel {

    // Modifier name: Accordion panel (when CSS transitions)
    //
    &--transition {
      ...

      // State: Accordion panel (when CSS transitions) / Hidden state
      //
      &[hidden] {
        ...
      }
    }
  }
}
```

**BEM CSS output:**
```css
.aria-accordion__panel--transition { ... }
.aria-accordion__panel--transition[hidden] { ... }
```

Don't you think `vh` as unit for `max-height` CSS property should be better? This will prevent infinity scroll and will limit extra long panel to fit with the viewport height if exceed. I am mainly thinking of mobile devices which have limited screen sizes.

Could you please commit your styles (taking into account my comments) on the [`feature/#2`](https://github.com/smillart/WAI-ARIA-Patterns-And-Widgets/tree/feature/%232) branch and directly in the following style sheet (expanded) for testing?
[./dist/accordion/css/accordion.min.css](https://github.com/smillart/WAI-ARIA-Patterns-And-Widgets/blob/feature/%232/dist/accordion/css/accordion.min.css)

Thanks

**Comment by laurent-b4y** [comment]
Hi,
Yes I know but for the module I have to override the default code, hence the very specific writing, just to be sure it'll work. I'll adapt it for the library so that it can easily be overridden.

Regarding the vh value for the max-height, I tried it but and it works well,  but the animation speed isn't consistent using vh, since that value is variable... and again for the purpose of the module the aim was to match the default animation as closely as possible. Extra long pannels will have a scroll-bar even when using vh, once the content is larger than the view port's height. Still if variation in animation speed is not an issue using vh is better.

**Comment by smillart** [comment]
This enhancement will be available with the next release 1.0.5.
